Gems/Atom/RPI/Code/Source/RPI.Public/Pass/ParentPass.cpp

```diff
@@ -479,6 +513,74 @@ namespace AZ
             return nullptr;
         }
 
+        void ParentPass::ClearMergeAsSubpassesFlag()
+        {
+            m_flags.m_mergeChildrenAsSubpasses = false;
+            for (auto& childPass : m_children)
+            {
+                childPass->m_flags.m_mergeChildrenAsSubpasses = false;
+            }
+        }
+
+        void ParentPass::CreateRenderAttachmentConfigurationForSubpasses()
+        {
+            if (!m_flags.m_mergeChildrenAsSubpasses)
+            {
+                // This is the most common scenario.
+                return;
+            }
+
+            // We validated already during AddChild() and InsertChild() that all children are
+            // of type RasterPass or subclass of RasterPass.
+            const auto numSubpasses = m_children.size();
+            AZ_Assert(numSubpasses > 1, "A parent pass that merges its children must have at least two children");
+
+            bool allChildrenBuiltLayout = true;
+            RHI::RenderAttachmentLayoutBuilder builder;
+            for (auto subpassIndex = 0;  subpassIndex < numSubpasses; subpassIndex++)
+            {
+                RasterPass* rasterChild = azrtti_cast<RasterPass*>(m_children[subpassIndex].get());
+                AZ_Assert(rasterChild != nullptr, "When merging subpasses, all children must be RasterPass");
+                auto* layoutBuilder = builder.AddSubpass(&rasterChild->GetPathName());
+                if (!rasterChild->BuildSubpassLayout(*layoutBuilder, subpassIndex))
+                {
+                    AZ_Error("ParentPass", false, "RasterPass [%s] failed to build its subpass layout.\n", rasterChild->GetName().GetCStr());
+                    allChildrenBuiltLayout = false;
+                    break;
+                }
+            }
+
+            if (!allChildrenBuiltLayout)
+            {
+                ClearMergeAsSubpassesFlag();
+                return;
+            }
+
+            allChildrenBuiltLayout = true;
```

Comment:
Can delete this line, allChildrenBuiltLayout must already be true after the above if( ) statement

Reply:
Line deleted.